packages/foam-core/src/model/uri.ts

```diff
@@ -0,0 +1,508 @@
+// Some code coming from
```

Comment:
We should append Microsoft's license statement as part of our LICENSE file:
https://github.com/microsoft/vscode/blob/main/LICENSE.txt